### ISSUE TITLE
question about <mnist_minimal_example> 

### ISSUE THREAD

**Issue by Johnson-yue** [VLL-HD/FrEIA#2]
Hi,  , thank you for your sharing , I am learning your method INN. and I can not understand the demo(**mnist_minimal_example**)  loss ? what is the meaning?? I have ever seen before.
![image](https://user-images.githubusercontent.com/10268274/61038759-23daae00-a400-11e9-8954-52f332ddae8c.png)

**Comment by Johnson-yue** [comment]
Hi, I know this nll loss compute is from Eq (6) in paper ,but where is L2 regularization? 

And Question 2:
        I run your experiments/mnist_minimal_example/train.py, and then run eval.py .Both run with default configure and plot those pictures...
![Screenshot from 2019-07-12 14-16-18](https://user-images.githubusercontent.com/10268274/61106591-34e2f800-a4b0-11e9-954d-90f9eac2b6c3.png)
I think the result is very bad . and how can I reimplement your experiments like your paper?

**Comment by ardizzone** [comment]
Hi, thank you for bringing this to my attention!
Concerning the L2 regularization: this is included in the pytorch optimizer, so we do not have to implement it by hand.

About the quality of results: You are right, they do not look good. I will investigate, and get back to you in a few days!

**Comment by ardizzone** [comment]
I could reproduce the issue on my end, and I also found the problem:
The amplitude for the noise augmentation was a factor of 10 off.
I just pushed commit e6bc8fd to fix it.
I also tweaked some other hyper-parameters, it now trains in half the time (15 minutes, 60 epochs).

If you want even better results, you can also increase the number of coupling blocks to 24, and train for 480 epochs (settings used in the paper, but takes several hours).

![2019-07-12_16-45-00](https://user-images.githubusercontent.com/33628695/61136955-ce1c0980-a4c4-11e9-8b03-a78b5e3079f9.png)

I will close the issue, but feel free to reopen if there are further problems.

**Comment by Johnson-yue** [comment]
@ardizzone  Hi, I also test for style transfer experiments like Fig.(7), so I modify the eval.py into [style_transfer.py](https://github.com/yqGANs/FrEIA/blob/master/style_transfer.py) and result like this:
![source-vs-rec](https://user-images.githubusercontent.com/10268274/61198144-1fb5d580-a70b-11e9-8f25-df848c785923.png)

I think in this configuration the cINN could not decompose into 10 essentially separate subnetworks.. because in this cause, some label is not generate image

**Comment by ardizzone** [comment]
Hi,

very interesting! For most of your samples it works fine, but as you said, some don't work at all.
This is just and issue of the training (and the model).

I just pushed the actual research code for the mnist task, used to produce the paper (some documentation and comments were still missing): 8ea016

I also included a model checkpoint.

The model from the paper is slightly larger and trained for longer, you can test the style transfer by running eval.py in the experiments/mnist_cINN folder.

I am confident it can be made much smaller and more efficient using convolutional layers, instead of all fully connected, I may update in the future (or you can experiment yourself).

**Comment by Johnson-yue** [comment]
@ardizzone Ok，thank you very much . when I test mnist experiments, I will check convolutional layers experiments like : colorization_minimal_experiment  and colorization_cINN.

Thanks for your sharing again. and I will learn more for this.
